Sources/Prometheus/Gauge.swift

```diff
@@ -0,0 +1,85 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftPrometheus open source project
+//
+// Copyright (c) 2018-2023 SwiftPrometheus project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftPrometheus project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Atomics
+import CoreMetrics
+
+public final class Gauge: Sendable {
+    let atomic = ManagedAtomic(Double.zero.bitPattern)
+
+    let name: String
+    let labels: [(String, String)]
+    let prerenderedExport: [UInt8]
+
+    init(name: String, labels: [(String, String)]) {
+        self.name = name
+        self.labels = labels
+
+        var prerendered = [UInt8]()
+        prerendered.reserveCapacity(64)
+        prerendered.append(contentsOf: name.utf8)
+        if let prerenderedLabels = Self.prerenderLabels(labels) {
+            prerendered.append(UInt8(ascii: "{"))
+            prerendered.append(contentsOf: prerenderedLabels)
+            prerendered.append(contentsOf: #"} "#.utf8)
+        } else {
+            prerendered.append(UInt8(ascii: " "))
+        }
+
+        self.prerenderedExport = prerendered
+    }
+
+    public func set(to value: Double) {
+        self.atomic.store(value.bitPattern, ordering: .relaxed)
+    }
+
+    public func increment(by amount: Double = 1.0) {
```

Comment:
A bit inconsistent, on purpose? The counter had Int64 variants of APIs as well, no need here?

Reply:
nope not un porpose